Applications/Spire/Include/Spire/Ui/DurationInputWidget.hpp

```diff
@@ -0,0 +1,61 @@
+#ifndef SPIRE_DURATION_INPUT_WIDGET_HPP
+#define SPIRE_DURATION_INPUT_WIDGET_HPP
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <QPainter>
+#include <QWidget>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  //! Displays a time duration input widget.
+  class DurationInputWidget : public QWidget {
+    public:
+
+      //! Signals that the time duration was modified.
+      using DurationSignal = Signal<void (boost::posix_time::time_duration)>;
+
+      //! Constructs a DurationInputWidget.
+      /*
+        \param parent The parent widget.
+      */
+      explicit DurationInputWidget(QWidget* parent = nullptr);
+
+      //! Sets the displayed time duration without trigger the duration
```

Comment:
Fix the typo/documentation. Also why doesn't this trigger the signal?

Reply:
The thought was that if something was setting the duration on the widget then it's dictating the value and hence already knows what the value is.

Reply:
Reasonable idea but widgets can be shared so the signal is necessary. In general all setters for a widget should trigger its corresponding signal (if one exists).